src/inmanta/parser/plyInmantaLex.py

```diff
@@ -132,7 +134,24 @@ def t_JCOMMENT(t: lex.LexToken) -> None:  # noqa: N802
 
 def t_MLS(t: lex.LexToken) -> lex.LexToken:
     r'"{3,5}([\s\S]*?)"{3,5}'
-    value = bytes(t.value[3:-3], "utf-8").decode("unicode_escape")
+
+    #   Check for the presence of an invalid escape sequence (ex: "\.")
+    #   Python < 3.12 raises a DeprecationWarning when encountering an invalid escape sequence
+    #   Python 3.12 will raise a SyntaxWarning
+    #   Future versions will eventually raise a SyntaxError
+    #   see ( https://docs.python.org/3.12/whatsnew/3.12.html#other-language-changes )
+    try:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("error", message="invalid escape sequence")
+            value: str = bytes(t.value[3:-3], "utf_8").decode("unicode_escape")
+    except DeprecationWarning:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            value = bytes(typing.cast(str, t.value)[3:-3], "utf_8").decode("unicode_escape")
+
+        msg: str = f"({t.lexer.inmfile}:{t.lexer.lineno}) Invalid escape sequence in multi-line string:\n{value}"
```

Comment:
I wonder if it adds value to have the string as part of the warning message. It can get quite verbose and you already have the line number. But I'm really bad at UX. What do you think?

Reply:
I have to agree that it will get ugly for long strings.